File: imiona/IMI/public/judge.cpp
#include "judge.h"
#include <iostream>
using namespace std;

namespace {

int mode;

int population;
int queryCount;
int nextQuery;
char names[32768][16];
char logins[65536][8];
char buf[32];

void init() {
	static bool doInit = true;
	if (doInit == false) return;
	doInit = false;
	cin >> mode;
	cin >> population >> queryCount;
	for (int i=0; i<population; ++i) {
		cin >> buf;
		for (int j=0; j<16; ++j) names[i][j] = buf[j];
	}
	for (int i=0; i<queryCount; ++i) {
		cin >> buf;
		for (int j=0; j<8; ++j) logins[i][j] = buf[j];
	}
} // init()

} // namespace

int Judge::getPopulation() {
	init();
	return population;
}

void Judge::getCitizenName(int id, char *name) {
	init();
	for (int i=0; i<16; ++i)
		name[i] = names[id][i];
}

int Judge::getQueryCount() {
	init();
	return queryCount;
}

void Judge::getCitizenLogin(char *login) {
	init();
	for (int j=0; j<8; ++j)
		login[j] = logins[nextQuery][j];
	++nextQuery;
}

void Judge::replyCitizenName(char *name) {
	init();
	for (int j=0; j<16; ++j)
		buf[j] = name[j];
	buf[16] = 0;
	cout << buf << "\n";
}

File: imiona/IMI/public/Judge.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;


public class Judge {
	static int mode;
	static int population = 0;
	static int queryCount = 0;
	static int nextQuery = 0;

	static char[][] names = new char[32768][16];
	static char[][] logins = new char[65536][8];
	static char[] buf = new char[32];


	static boolean doInit = true;
	static Scanner s;
	static void init() {
		if (doInit == false) return;
		doInit = false;
		s = new Scanner(System.in);
		mode = s.nextInt();

		population = s.nextInt();
		queryCount = s.nextInt();
		s.nextLine();
		for (int i=0; i<population; ++i) {
			String line = s.nextLine();
			for (int j=0; j<16; ++j) names[i][j] = line.charAt(j);
		}
		for (int i=0; i<queryCount; ++i) {
			String line = s.nextLine();
			for (int j=0; j<8; ++j) logins[i][j] = line.charAt(j);
		}
	} // init()

	static public int getPopulation() {
		init();
		return population;
	}

	static public void getCitizenName(int id, char[] name) {
		init();
		for (int j=0; j<16; ++j)
			name[j] = names[id][j];
	}

	static public int getQueryCount() {
		init();
		return queryCount;
	}

	static public void getCitizenLogin(char[] login) {
		init();
		for (int j=0; j<8; ++j)
			login[j] = logins[nextQuery][j];
		++nextQuery;
	}

	static public void replyCitizenName(char[] name) {
		init();
		for (int i=0; i<16; ++i) System.out.print(name[i]);
		System.out.println("");
	}
}

